### PR TITLE
added post for incremental dom

### PR DESCRIPTION
**Pull request by tehsis**
## Pendings
* [x] Define post's image
* [x] Make repos and examples are in the correct github account and published on npm

Reviewed by @mgonto and @chenkie

### CODE REVIEW

**Comment by mgonto** [comment]
This needs the picture of how it looks live @tehsis

File: _posts/2015-09-25-incremental-dom.markdown [hunk 1]
```diff
@@ -0,0 +1,327 @@
+---
+layout: post
+title: "Incremental DOM"
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: View Fight Series: Virtual DOM vs Glimmer vs Incremental DOM.  Put that as a title

File: _posts/2015-09-25-incremental-dom.markdown [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+description: Learn about Incremental DOM and how to work with it
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: Learn how Incremental DOM works and how it compares to Glimmer and Virutal DOM

File: _posts/2015-09-25-incremental-dom.markdown [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+date: 2015-09-25 00:47
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: Change the date before merge

File: _posts/2015-09-25-incremental-dom.markdown [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+author:
+  name: Pablo Terradillos
+  url: https://twitter.com/tehsis
+  mail: tehsis@auth0.com
+  avatar: https://s.gravatar.com/avatar/647b1eea820b3fc8a5aee0383930b888
+design:
+  bg_color: "#232228"
+  image: https://cdn.auth0.com/docs/img/incremental-post-picture.png
+tags:
+- virtual DOM
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: ember
view

**Comment by mgonto** [comment]
Did you send it to wordy? Send it to wordy!

File: _posts/2015-09-25-incremental-dom.markdown [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+
+Ember's implementation is called _Glimmer_, and it is a substantial improvement to React's. Instead of building a complete representation of the _DOM Tree_, it just takes the elements from a Handlebars template that are dynamic and builds a _tree of streams_, reducing the amount of memory needed to keep them.
+
+Google is working on a different approach with [Incremental DOM](https://github.com/google/incremental-dom) (IDOM). We can say that it has the same base as _virtual DOM_, but with a different approach: **instead of building a representation of the _DOM tree_ in memory, it just uses the real DOM to diff against new trees.**
+
+While React provides JSX as a template engine and Glimmer relies on HTMLBars, **IDOM** is just a simple API for building DOM elements and is not intended to be used directly but as a compilation target for template engines.
+
+In this article, we're going to explore the simple Incremental DOM API by building a library to work with it, along with a functional to-do app.
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: Add a graphic above this `In this article` that shows the differences on how Virutal DOM, Incremental DOM and Glimmer work memory wise :). @iamevilrabbit can help with making it beautufil after you make it.

File: _posts/2015-09-25-incremental-dom.markdown [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+
+```js
+patch(document.body, renderHello);
+```
+
+And that's it! All you need to know is that `patch` receives an extra parameter that will be passed as an argument of the render function, and you can say to all your coulleages that you know Incremental DOM.
+
+That was a simple example, but the real power comes when we have a DOM structure that constantily needs to be changed. Lets build a _real app_ to see how it works.
+
+## Making IDOM more friendly
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: friendlier :D

File: _posts/2015-09-25-incremental-dom.markdown [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+  let idx = LiEl.dataset.idx;
+  this.removeItem(idx);
+});
+```
+
+Feel free to have a look at the [final implementation](http://auth0.github.io/incremental-todo/) of the todo app.
+
+Fire up your inspector and note that even though we're telling our component to re-render itself every time an item is added or deleted, IDOM takes care of re-rendering just the item that changes:
+
+![](https://cdn.auth0.com/docs/img/incremental-dom-inspector.gif)
```
Comment (mgonto) [inline _posts/2015-09-25-incremental-dom.markdown]: Add all of the alts

**Comment by mgonto** [comment]
Add an aside that references to Auth0.

Say that Auth0 is investing a lot of time in View Engines and analyzing which one is better. Say that we'fe done Lock Passwordless with React and that we're planning ont rying out Incremental Dom in prod soon. Also say that we have an SDK for React (link) and one for Incremental DOM coming soon.

**Comment by mgonto** [comment]
@tehsis how are we with the updates?

**Comment by tehsis** [comment]
@mgonto Updates are finished.

I'm just not sure about the image you're suggesting.

**Pull request merged**